modules/battery_manager/php/battery_manager.class.inc

```diff
@@ -0,0 +1,245 @@
+        $instruments = $this->makeKeySameAsValue(\Utility::getAllInstruments());
+
+        $stages = $this->getStageList();
+
+        $subprojects = $this->makeKeySameAsValue(\Utility::getSubprojectList(null));
+
+        $visits = \Utility::getVisitList();
+
+        $sites = $this->makeKeySameAsValue(\Utility::getSiteList(false));
+
+        $firstVisit = $this->getYesNoList();
+
+        $active = $this->getYesNoList();
+
+        // Form Elements
+        $this->addSelect('instrument', 'Instrument', $instruments);
+        $this->addBasicText('minimumAge', 'Minimum age');
+        $this->addBasicText('maximumAge', 'Maximum age');
+        $this->addSelect('stage', 'Stage', $stages);
+        $this->addSelect('subproject', 'Subproject', $subprojects);
+        $this->addSelect('visitLabel', 'Visit Label', $visits);;
+        $this->addSelect('site', 'Site', $sites);
+        $this->addSelect('firstVisit', 'First Visit', $firstVisit);
+        $this->addBasicText('instrumentOrder', 'Instrument Order');
+        $this->addSelect('active', 'Active', $active);
+
+        $this->tpl_data["hasWritePermission"] = $this->hasWritePermission;
+
+        return true;
+    }
+
+    /**
+     * Return associative array of stages
+     *
+     * @return array
+     */
+    function getStageList()
+    {
+        return array(
+                "Not Started"   => 'Not Started',
+                "Screening"     => 'Screening',
+                "Visit"         => 'Visit',
+                "Approval"      => "Approval",
+                "Subject"       => "Subject",
+                "Recycling Bin" => "Recycling Bin",
+               );
+    }
+
+    /**
+     * Return associative array of yes and no values
+     *
+     * @return array
+     */
+    function getYesNoList()
+    {
+        return array(
+                'Y' => 'Yes',
+                'N' => 'No',
+               );
+    }
+
+    /**
+     * Change the associative array so that the keys are the same as the values
+     * (ensures filter will work on columns displaying full names)
+     *
+     * @param array $list that needs to be updated
+     *
+     * @return array
+     */
+    function makeKeySameAsValue($list)
+    {
+        $options = array();
+        foreach ($list as $key => $value ) {
+            $options[$value] = $value;
+        }
+        return $options;
+    }
+
+    /**
+     * Build a list of test battery records to display in Data Table
+     *
+     * @return bool
+     * @throws DatabaseException
+     */
+    function _setupVariables()
+    {
+        $user =& \User::singleton();
+        // the base query
+        $query  = " FROM test_battery b
+                   LEFT JOIN test_names t ON b.Test_name = t.Test_name
+                   LEFT JOIN subproject s ON b.SubprojectID = s.SubprojectID
+                   LEFT JOIN psc p ON b.CenterID = p.CenterID ";
+        $query .= " WHERE 1=1 ";
+
+        // set the class variables
+        $this->columns
+            = [
```

Comment:
consensus is to user `array(...)` if you found this in another class, the other class should be amended in a separate PR